### ISSUE TITLE
Move to SI units - add polarization

### ISSUE THREAD

**Issue by Alexboiboi** [magpylib/magpylib#698]
In line with version 5 a new polarization parameter should be added to magnets as well as moving the entire library to SI units.

-  [x] add `polarization` parameter.
  - [x] `magnetization` and `polarization` should be two co-dependent parameters (setting one should set the other automatically)
- [x] Move to SI units - SI unit input should return SI unit output.
  - [x] Inputs 
    - [x] `magnetization`  → _A/m_
    - [ ] `polarization`  → _T_
  - [x] Outputs 
    - [x] `B-field`  → _T_
    - [x] `H-field`  → _A/m_
- [x] Other places of units changes
  - [x] docstrings
  - [ ] input checks
  - [x] graphics
  - [ ] objects `__repr__`
- [x] change argument order for all classes:
     ```python
     def __init__(self, position=(0, 0, 0), orientation=None, <...>, style=None, **kwargs)
     ```
  - [x] Magnets:
    - [x] Cuboid, Cylinder, CylinderSegment
       ```python
        <dimension=None,  polarization=None, magnetization=None>
       ```
    - [x] Sphere
       ```python
        <diameter=None,  polarization=None, magnetization=None>
       ```
    - [x] Tetrahedron, Triangle
       ```python
        <vertices=None,  polarization=None, magnetization=None>
       ```
    - [x] TriangularMesh
       ```python
        <
          position=(0, 0, 0),
          orientation=None,
          vertices=None,
          faces=None,
          polarization=None,
          magnetization=None,
          check_open="warn",
          check_disconnected="warn",
          check_selfintersecting="warn",
          reorient_faces="warn",
       >
       ```
  - [x] Currents
    - [x] Circle
       ```python
       <diameter=None, current=None>
       ```
    - [x] Polyline
       ```python
       <vertices=None, current=None>
       ```
  - [x] Misc
    - [x] Dipole
       ```python
       <moment=None>
       ```
    - [x] CustomSource
       ```python
       <field_func=None>
       ```

**Comment by OrtnerMichael** [comment]
What happening on the core level:

**NEW ARGUMENT ORDERS**
    - field
    - observers
    - position=(0, 0, 0),
    - orientation=None,
    - dimension / diameter / vertices / ...
    - polarization / current / moment
    - magnetization
    - style=None,
    - **kwargs,

**CHANGES IN CORE FUNCTIONS:**
- comment #CORE
- *, no more positionals
- magnetization <> polarizations
- variable order reflecting new order
- variable names in singular (again) - we can change this later if we want

- Top line in plural
- SI units are used for all inputs and outputs.
- rearrange Parameters
- everything SI: T, A/m and m

- fix computation to SI output: B is good, H=B/mu0
    - include MU0
- implement/fix tests
    - add self-consistency test B = MU0*H + J

- fix doc examples
- Notes: advanced units

**Comment by Alexboiboi** [comment]
fixed by #712
